src/ast/passes/resource_analyser.cpp

```diff
@@ -358,20 +358,8 @@ void ResourceAnalyser::visit(Call &call)
       resources_.str_buffers++;
   }
 
-  // Aggregation functions like count/sum/max are always called like:
-  //   @ = count()
-  // Thus, we visit AssignMapStatement AST node which visits the map and
-  // assigns a map key buffer. Thus, there is no need to assign another
-  // buffer here.
-  //
-  // The exceptions are:
-  // 1. lhist/hist/tseries because the map key buffer includes both the key
-  //    itself and the bucket ID from a call to linear/log2/tseries functions.
-  // 2. has_key/delete because the map key buffer allocation depends on
-  //    arguments to the function e.g.
-  //      delete(@, 2)
-  //    requires a map key buffer to hold arg1 = 2 but map.key_expr is null
-  //    so the map key buffer check in visit(Map &map) doesn't work as is.
+  // lhist/hist/tseries need an additional key for the bucket ID from a call to
+  // linear/log2/tseries functions.
   if (call.func == "lhist" || call.func == "hist" || call.func == "tseries") {
```

Comment:
Can you include an explanation of the general requirement here? I understand that the previous comment was a bit off, but it seems more *half-right* and I don't think we should lose it completely while it is still relevant.

Reply:
Done. Wrote up something more detailed.